db/models.py

```
@@ -49,4 +53,51 @@
     )
 
     def __str__(self) -> str:
-        return f"{self.movie.title} {str(self.show_time)}"
+        return f"<{self.movie.title} {str(self.show_time)}>"
+
+    class Order(models.Model):
+        created_at = models.DateTimeField(default=timezone.now)
+        user = models.ForeignKey("User", on_delete=models.CASCADE)
+
+        class Meta:
+            ordering = ["-created_at"]
+
+        def __str__(self) -> str:
+            return f"{self.created_at}"
+
+    class Ticket(models.Model):
+        movie_session = models.ForeignKey(MovieSession, on_delete=models.CASCADE)
+        order = models.ForeignKey(Order, on_delete=models.CASCADE)
+        row = models.IntegerField()
+        seat = models.IntegerField()
+
+        def __str__(self) -> str:
+            return (f"{self.movie_session.movie.title} "
+                    f"{self.movie_session.show_time} "
+                    f"(row: {self.row}, seat: {self.seat})")
+
+        def clean(self) -> bool:
+            if self.row > self.movie_session.cinema_hall.rows:
+                raise ValidationError({
+                    "row": ["row number must be in available "
+                            "range: (1, rows): (1, 10)"]
```

The `clean()` method uses hardcoded values (10 and 12) for rows and seats in error messages. It should use the actual values from `self.movie_session.cinema_hall.rows` and `self.movie_session.cinema_hall.seats_in_row`. This violates: 'Raise ValidationError if out of range, with messages using the actual values.'